app/utils/service_client/middleware/request_id.rb

```diff
@@ -0,0 +1,11 @@
+module ServiceClient
+  module Middleware
+    class RequestID < MiddlewareBase
+      def enter(ctx)
+        headers = ctx.fetch(:req).fetch(:headers)
```

Comment:
fetch doesn't buy much here because the next line already expected :reg to be defined. I feel it's good to let middleware like this fail when ctx is not of expected shape. That can only mean a configuration error (middleware is incorrectly ordered or missing) so it's good to expose it in the dev phase as early as possible.